modules/llm-cache/ds/kv_state_cache_manager.cc

```diff
@@ -76,13 +76,31 @@ KVStateCacheManager::KVStateCacheManager(int dimension, int cacheCapacity,
     }
   }
 
-  // syncThread = new std::thread(threadFunc);
   syncThread = new std::thread(SyncThreadFunc, this);
+  if (syncThread == nullptr) {
+    throw std::runtime_error("Create sync thread failed!");
+  }
 
   // TBD
   // use lease to prevent the deadlock if the client is down
 }
 
+Status KVStateCacheManager::MakeKVStateCacheManager(
+    KVStateCacheManager*& manager, int dimension, int cacheCapacity, int layer,
+    int blockSize, int syncInterval, std::string socket) {
+  try {
+    manager = new KVStateCacheManager(dimension, cacheCapacity, layer,
+                                      blockSize, syncInterval, socket);
+    if (manager == nullptr) {
+      return Status::NotEnoughMemory("Failed to make KVStateCacheManager.");
+    }
```

Comment:
I believe the `new` operator (without `-fno-expection`) never returns `nullptr`, rather, it throws `std::bad_alloc`.

Reply:
Ok, I understand.